### ISSUE TITLE
Spot redundant return

### ISSUE THREAD

**Issue by ndmitchell** [ndmitchell/hlint#473]
```
forM_ ...
return ()
```

```
void $ forM_ ...
```

Both should be warned on (and if either uses `forM` it should also be warned on).

**Comment by ndmitchell** [comment]
Done, for all types of unit returning function.
